azurerm/internal/services/network/tests/loadbalancer_rule_resource_test.go

```diff
@@ -340,6 +340,41 @@ func TestAccAzureRMLoadBalancerRule_disappears(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMLoadBalancerRule_vmssBackendPool(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_lb_rule", "test")
+	var lb network.LoadBalancer
+	lbRuleName := fmt.Sprintf("LbRule-%s", acctest.RandStringFromCharSet(8, acctest.CharSetAlpha))
+
+	subscriptionID := os.Getenv("ARM_SUBSCRIPTION_ID")
```

Comment:
rather than using the environment variable please use this from `data.Client().SubscriptionID`

Reply:
I just copied it from the other test functions, I'll change them all.